pkg/coscheduling/coscheduling.go

```diff
@@ -55,15 +55,31 @@ var _ framework.PostBindPlugin = &Coscheduling{}
 
 const (
 	// Name is the name of the plugin used in Registry and configurations.
-	Name = "Coscheduling"
+	Name                                       = "Coscheduling"
+	defaultPermitWaitingTimeSeconds      int64 = 60
+	defaultDeniedPGExpirationTimeSeconds int64 = 20
 )
 
-// New initializes a new plugin and returns it.
+// New initializes and returns a new Coscheduling plugin.
 func New(obj runtime.Object, handle framework.FrameworkHandle) (framework.Plugin, error) {
-	args, ok := obj.(*config.CoschedulingArgs)
-	if !ok {
-		return nil, fmt.Errorf("want args to be of type CoschedulingArgs, got %T", obj)
+	var (
+		args *config.CoschedulingArgs
+		ok   bool
+	)
+	if obj != nil {
```

Comment:
Any particular reason checking  if it's nil?

Reply:
It skips setting the defaults  if the configuration is empty.  If the configuration cannot be empty, we can just return with an error. Again, if we use the api version `v1beta1,` we won't need it.